pkg/cue/cuex/compiler.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2023 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cuex
+
+import (
+	"github.com/kubevela/pkg/cue/cuex"
+
+	"github.com/oam-dev/kubevela/pkg/cue/cuex/providers/config"
+)
+
+// KubeVelaDefaultCompiler compiler for cuex to compile
+var KubeVelaDefaultCompiler = cuex.NewCompilerWithInternalPackages(
```

Comment:
Let's use singleton.New to allow lazy initialization. And it can moved to utils/common/cuex.go so that when importing this KubeVelaDefaultCompiler and cuex package in kubevela/pkg, developer does not need to rename the package to pkgcuex.